### ISSUE TITLE
"Export notes" file doesn't have .json extension

### ISSUE THREAD

**Issue by tom93** [FossifyOrg/Notes#13]
### Checklist

- [ ] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Notes/releases/latest). *N/A (not released yet)*
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Notes/issues) or [closed](https://github.com/FossifyOrg/Notes/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Notes/discussions) or [closed](https://github.com/FossifyOrg/Notes/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Notes/blob/master/CONTRIBUTING.md).

### Affected app version

SimpleMobileTools 6.17.0 (since FossifyOrg Notes hasn't been released yet)

### Affected Android/Custom ROM version

~~All~~
Android 9
Happens both when completing the save action using the native Files app and when using Fossfy File Manager.

### Affected device model

~~All~~
Android-x86 VM
Also a physical device

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

In Settings, click "Export notes".
Observe that the prompt says "Filename (without .json)" and the pre-populated filename does not have a .json suffix.
Click OK. Pick a destination folder.

### Expected behavior

The exported file should have a .json file extension.

### Actual behavior

The exported file does not have a .json file extension.

### Screenshots/Screen recordings

<img alt="screenshot-1" src="https://github.com/FossifyOrg/Notes/assets/5887562/1b3eae22-34c5-41a0-a155-f083e3a4dae8" width=528 />
<img alt="screenshot-2" src="https://github.com/FossifyOrg/Notes/assets/5887562/59bb1b29-fe1a-4c87-b6a0-c18e8835dc0f" width=528 />
<img alt="screenshot-3" src="https://github.com/FossifyOrg/Notes/assets/5887562/e600997f-7b99-43ff-9b45-ebdf8b774715" width=528 />
<img alt="screenshot-4" src="https://github.com/FossifyOrg/Notes/assets/5887562/0b51e0e7-24dd-4bbb-b4e5-f16f118b362a" width=528 />
<img alt="screenshot-5" src="https://github.com/FossifyOrg/Notes/assets/5887562/105fcfd3-aa4c-42c5-98e5-c529d1c1bac3" width=528 />

### Additional information

Fixing this is ~~trivial~~ *hopefully trivial*, but also raises the question of what should be changed:

**A.** Obvious fix: Leave the UI as is, and automatically append ".json" when saving. (This is consistent with the automatic backup feature.) Implemented in #14.

**B.** Change the UI so the prompt is just "Filename" and the initial value includes a ".json" suffix. I like this, because it's simple and direct -- it shows the user that the output will be a .json file without the user having to stop and think (thought process: "What's the output format going to be? Wait, the tiny prompt says "without .json", meaning I don't write .json, meaning *the app will probably automatically add .json*, so the output is probably .json"). Also allows the user to easily pick a different file extension (e.g. .txt to allow viewing in text editor). However, this is inconsistent with the automatic backup and most of the other apps (e.g. the Calendar export prompt says "without .ics" and automatically appends .ics).

**Comment by Aga-C** [comment]
Have you tested in on multiple real devices? Writing "All" under "Affected Android version" and "Affected device model" is a very bold statement. You should always write on what device you've really encountered it.

I think, it may only affect specific Android versions. On Android 10 (Xiaomi Redmi 7A) and Android 13 (OnePlus 8 Pro, OnePlus 9 Pro) it saves with `.json` extension. However, I was able to reproduce it on Android 6 (OnePlus 2). 

If someone were to fix this, there should be no UI change because similar UI is currently also in other apps. It should only fix that save always saves with proper extension.

**Comment by tom93** [comment]
Oops, I added device details (I did test on both a VM and a physical device, and thought it was just a trivial typo, but evidently I was wrong). I guess the file extension is added automatically, either on newer versions of Android or by the file manager app used to pick the destination.

Can you please check that my proposed fix in #14 doesn't break things on the working devices that you've listed? (I'm worried about ending up with .json.json as the file extension, but I'm not able to test on those Android versions myself.)

**Comment by p4bl0-** [comment]
I don't mind it but can confirm that the .json extension is missing on my install (v6.17 of the app on Android 13).

**Comment by tom93** [comment]
I did more testing and my conclusion is that the native Files app automatically adds the extension since Android 10 (tested on Android 9, 10, 14). Fossify File Manager never adds an extension (tested on Android 9, 10).

So I think the simplest solution is to always add the extension in the Notes app. It fixes the missing extension issue, and as far as I can tell it has no negative consequences -- it doesn't cause double extensions, and for duplicates, the versions of the Files app that can add the extension automatically (Android 10+) correctly insert the "(1)" before the extension even when the extension was added by the Notes app.
